pkg/services/ngalert/ngalert.go

```diff
@@ -410,7 +410,7 @@ func configureHistorianBackend(ctx context.Context, cfg setting.UnifiedAlertingS
 		testConnCtx, cancelFunc := context.WithTimeout(ctx, 10*time.Second)
 		defer cancelFunc()
 		if err := backend.TestConnection(testConnCtx); err != nil {
-			return nil, fmt.Errorf("failed to ping the remote loki historian: %w", err)
+			l.Error("Failed to communicate with configured remote Loki backend, state history may not be persisted", "error", err)
```

Comment:
Should this error differentiate between a connection error because the configuration is incorrect compared to when the Loki server is unavailable?

Reply:
It already does, above on line 403. The config gets validated when the client is constructed. If the config is bad, we fail the initialization.

The only errors returned here are ones that happen in the process of actually executing the request. Could not reach host, timeouts, 404s, failed to authenticate... things like that.

Reply:
If you're asking about differentiating between timeouts vs things like auth failures - I think we should rather lump all "client" type errors together. We never know what's happening on the Loki side, or whether the software behind the configured endpoint even is loki in the first place - so these things can change without any change in Grafana config. 

Let's default to keeping Grafana running, but only disabling state history, as state history data isn't quite as critical as the overall health of the system.

Reply:
Sorry what I meant was the configuration is valid from the perspective of the client, but rejected by the Loki server because the credentials are incorrect. It suppose with these logs a user can still figure out their history doesn't work, but they will just need to observe missing history first before they realize?

Reply:
Yup, and we 100% do not want to bring down the alerting system or all of Grafana in this case - this is by design.

State history data is less important than rules and notifications, and we do not want to bind grafana's ability to run to the behavior of a non-required external system.